OpenRA.Game/Network/OrderManager.cs

```diff
@@ -179,8 +179,8 @@ public void Tick()
 			if (!IsReadyForNextFrame)
 				throw new InvalidOperationException();
 
-			Connection.Send(NetFrameNumber + FramesAhead, localOrders.Select(o => o.Serialize()).ToList());
-			localOrders.Clear();
+			Connection.Send(NetFrameNumber + FramesAhead, localOrders.Where(o => !o.IsImmediate).Select(o => o.Serialize()).ToList());
+			localOrders.RemoveAll(o => !o.IsImmediate);
```

Comment:
Well derp, the old code was indeed obviously wrong.

IMO the bookkeeping to manage this is now less than ideal. We have one place that unconditionally writes into `localOrders` but then have to walk the list four times checking the `IsImmediate` flag.

It would be cleaner for `IssueOrder` to check `IsImmediate` once and then write to either `localOrders` or a new `localImmediateOrders` list.  `Tick` and `TickImmediate` can then unconditionally consume and then clear their respective lists.